app/plugins/rwahsNavigation/rwahsNavigationPlugin.php

```diff
@@ -0,0 +1,194 @@
+            $pa_nav_info['find']['navigation'] = $this->_getCustomSearchMenuItems() + $pa_nav_info['find']['navigation'];
+            unset($pa_nav_info['find']['navigation']['objects']);
+        }
+        return $pa_nav_info;
+    }
+
+    private function _getSpacer() {
+        return array(
+            'displayName' => '<div class="sf-spacer"></div>'
+        );
+    }
+
+    private function _getCustomNewMenuItems() {
+        $va_custom_items = array();
+        $va_shortcuts = $this->opo_config->get('new_menu_shortcuts');
+        if (is_array($va_shortcuts)) {
+            foreach ($va_shortcuts as $vs_key => $vs_type_code) {
+                $vo_type = ca_list_items::find(
+                    array( 'idno' => $vs_type_code ),
+                    array( 'returnAs' => 'firstModelInstance' )
+                );
+                if (!$vo_type) {
+                    // Don't add a menu item if the type code does not resolve correctly.
+                    continue;
+                }
+                $va_custom_items[$vs_key] = array(
+                    'displayName' => $vo_type->get('preferred_labels'),
+                    'requires' => array(
+                        'action:can_create_ca_objects' => 'AND',
+                        'configuration:!ca_objects_disable' => 'AND'
+                    ),
+                    'default' => array(
+                        'module' => 'editor/objects',
+                        'controller' => 'ObjectEditor',
+                        'action' => 'Edit'
+                    ),
+                    'parameters' => array(
+                        'type_id' => 'string:' . $vo_type->getPrimaryKey()
+                    )
+                );
+            }
+        }
+        $va_custom_items['spacer'] = $this->_getSpacer();
+        return $va_custom_items;
+    }
+
+    private function _getCustomSearchMenuItems() {
+        $va_custom_items = array();
+        $va_shortcuts = $this->opo_config->get('search_menu_shortcuts');
+        if (is_array($va_shortcuts)) {
+            foreach ($va_shortcuts as $vs_key => $va_shortcut) {
+                if (!isset($va_shortcut['type_code']) || !isset($va_shortcut['form_code']) || !isset($va_shortcut['display_code'])) {
+                    // Don't even try to look up the type, form or display if any is unset.
+                    continue;
+                }
+                $vo_type = ca_list_items::find(
+                    array( 'idno' => $va_shortcut['type_code'] ),
+                    array( 'returnAs' => 'firstModelInstance' )
+                );
+                $vn_form_id = ca_search_forms::find(
+                    array( 'form_code' => $va_shortcut['form_code'] ),
+                    array( 'returnAs' => 'firstId' )
+                );
+                $vn_bundle_display_id = ca_bundle_displays::find(
+                    array( 'display_code' => $va_shortcut['display_code'] ),
```

Comment:
Well I looked at the the schema, and `display_code` and `form_code` are both unique, so you don't need to add `table_num` in here or above (although that uniqueness is not restricted in `profile.xsd`).